lib/controllers/project.js

```diff
@@ -21,779 +22,720 @@
-    ],
-    function(err, domain) {
-      if (!err) {
-        callback(null, domain);
-      } else {
-        callback(err);
-      }
+    }
+  ],
+    function (err, domain) {
+      if (err) return callback(err);
+      return callback(null, domain);
     }
   );
 };
 
-//-----------------------------------------------------------------------------
-Project.prototype._deploy = function(projectId, file, registry, withTests, callback) {
-  modulus.io.print('Uploading project...');
-  var status = Project.status.uploading;
+// ----------------------------------------------------------------------------
+Project.prototype._deploy = function (projectId, file, registry, withTests, callback) {
+  var ubar, client, options, status = Project.status.uploading;
+  print('Uploading project...');
 
   // start checking for status changes
-  // show progress for upload and indeterminate progress for deploying
-  var ubar = new ProgressBar('Upload progress [:bar] :percent', {
+  // show progress for upload and Indeterminate progress for deploying
+  ubar = new ProgressBar('Upload progress [:bar] :percent', {
     complete: '=',
     incomplete: ' ',
     width: 20,
     total: 100
   });
-  if (process.stdout.isTTY) { ubar.tick(); }
 
-  var client = new Uploader(),
-    options = {
-      registry: registry,
-      withTests: !!withTests
-    };
-
-  client.upload(projectId, file, options);
+  if (process.stdout.isTTY) ubar.tick();
 
-  client.on('progress', function(progress) {
+  client = new Uploader();
+  options = { registry: registry, withTests: Boolean(withTests) };
 
+  client.upload(projectId, file, options);
+  client.on('progress', function (progress) {
     // The progress bar UI starts at 1. This keeps it from jumping
     // to zero, then back to 1.
-    if (progress < 0.01) {
-      progress = 0.01;
-    }
-
-    if (process.stdout.isTTY) { ubar.tick((progress * 100) - ubar.curr); }
+    if (progress < PROGRESS_MIN) progress = PROGRESS_MIN;
+    if (process.stdout.isTTY) ubar.tick(progress * PROGRESS_MAX - ubar.curr);
   });
 
-  client.on('error', function(err) {
-    //closes the ubar readline interface
+  client.on('error', function (err) {
+    // closes the ubar readline interface
     ubar.rl.close();
-
     callback(err);
   });
 
-  client.on('end', function() {
-
-    var logLength = 0;
-    var getDeployLogs = function(cb) {
-      librarian.project.getDeployLogs(projectId, userConfig.data.apiKey, function(err, result) {
-        for(var key in result) {
-          var log = result[key];
-          modulus.io.write(log.substring(logLength));
-          logLength = log.length;
-          break;
-        }
+  client.on('end', function () {
+    var deployLogs, getDeployLogs, projectStatus, logLength = 0;
+    getDeployLogs = function (cb) {
+      Librarian.project.getDeployLogs(
+        projectId, UserConfig.data.apiKey, function (err, result) {
+          var key, log;
+          if (err) return callback(Errors.getMessage(err));
```

Comment:
Do you want me to disable the eslint error that is throwing for the un-handled error and make a TODO? or do you have another suggestion?